src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1124,7 +1142,8 @@ private Binder GetFieldOrPropertyInitializerBinder(FieldSymbol symbol, Binder ou
 
         private static bool IsMemberDeclaration(CSharpSyntaxNode node)
         {
-            return (node is MemberDeclarationSyntax) || (node is AccessorDeclarationSyntax) || (node is AttributeSyntax);
+            return (node is MemberDeclarationSyntax) || (node is AccessorDeclarationSyntax) || 
```

Comment:
We would have to check multiple kinds for each class, it is more efficient to do a type check in this case.